packages/block-editor/src/components/inserter/index.js

```diff
@@ -105,10 +119,67 @@ class Inserter extends Component {
 
 export default compose( [
 	withSelect( ( select, { rootClientId } ) => {
-		const { hasInserterItems } = select( 'core/block-editor' );
-
+		const {
+			hasInserterItems,
+			__experimentalHasOnlyOneAllowedBlockType,
+			__experimentalGetTheOnlyAllowedBlockType,
+		} = select( 'core/block-editor' );
+		const allowedBlock = getBlockType( __experimentalGetTheOnlyAllowedBlockType( rootClientId ) );
 		return {
 			hasItems: hasInserterItems( rootClientId ),
+			hasOnlyOneAllowedInserterItem: __experimentalHasOnlyOneAllowedBlockType( rootClientId ),
+			blockTitle: allowedBlock ? allowedBlock.title : '',
+		};
+	} ),
+	withDispatch( ( dispatch, ownProps, { select } ) => {
+		return {
+			insertTheOnlyAllowedItem: () => {
+				const { rootClientId, clientId, isAppender, destinationRootClientId } = ownProps;
+				const {
+					__experimentalHasOnlyOneAllowedBlockType,
+					__experimentalGetTheOnlyAllowedBlockType,
+				} = select( 'core/block-editor' );
+
+				const hasOnlyOneAllowedInserterItem = __experimentalHasOnlyOneAllowedBlockType( rootClientId );
+
+				if ( ! hasOnlyOneAllowedInserterItem ) {
+					return false;
+				}
+
+				const parentAllowedBlocks = __experimentalGetTheOnlyAllowedBlockType( rootClientId );
```

Comment:
My understanding is that the function should only return one block? If so, the variable name shouldn't be plural.